docs/express.asciidoc

```diff
@@ -0,0 +1,250 @@
+})
+
+// any errors caught by Express can be logged by the agent as well
+app.use(apm.middleware.express())
+
+app.listen(3000)
+----
+
+The agent will now monitor the performance of your Express application and record any uncaught exceptions.
+
+[[express-advanced-configuration]]
+==== Advanced configuration
+
+In the above example we initialize the agent by calling the <<apm-start,`start()`>> function.
+This function takes an optional options object used to configure the agent.
+Any option not supplied via the options object can instead be configured using environment variables.
+So if you prefer, you can set the same configuration options using environment variables:
+
+[source,bash]
+----
+ELASTIC_APM_APP_NAME=<app name>
+ELASTIC_APM_SECRET_TOKEN=<token>
+ELASTIC_APM_SERVER_URL=<server url>
+----
+
+And then just start the agent like so:
+
+[source,js]
+----
+// Start the agent before any thing else in your app
+var apm = require('elastic-apm').start()
+----
+
+See all possible ways to configure the agent <<configuring-the-agent,in the API documentation>>.
+
+[[express-full-documentation]]
+==== Full documentation
+
+* <<setup,Setup and Configuration>>
+* <<api,API Reference>>
+
+[[express-performance-monitoring]]
+=== Performance monitoring
+
+Elastic APM automatically measures the performance of your Express application.
+It records traces for database queries,
+external HTTP requests,
+and other slow operations that happens during requests to your Express app.
+
+By default the agent will trace <<compatibility,the most common modules>>.
+To trace other events,
+you can use custom traces.
+For information about custom traces,
+see the <<custom-traces,Custom Traces section>>.
+
+Traces are grouped in transactions - by default one for each incoming HTTP request.
+But it's possible to create custom transactions not associated with an HTTP request.
+See the <<custom-transactions,Custom Transactions section>> for details.
+
+[[express-unknown-routes]]
+==== Unknown routes
+
+When viewing the performance metrics of your application in Elastic APM,
+you might see some transactions named "unknown route".
+This indicates that the agent detected an incoming HTTP request to your application,
+but didn't know which route in your Express app the HTTP request matched.
+
+This might simply be 404 requests,
+which by definition doesn't match any route,
+or it might be a symptom that the agent wasn't installed correctly.
+If you see this or can't get any meaningful metrics to show up,
+please follow the <<troubleshooting,Troubleshooting Guide>>.
+
+[[express-error-logging]]
+=== Error logging
+
+The Node.js agent will by default watch for uncaught exceptions and log these with Elastic APM.
+But in most cases you also want to log non-thrown errors.
+This section explains how.
+
+[[express-error-handlers]]
+==== Express error handlers
+
+Express works by funneling all incoming http requests through a series of middlewares (one of which being your router).
```

Comment:
again, I've never seen middlewares written as plural. Maybe say "a series of middleware components".